src/components/AuthCheck.tsx

```diff
@@ -1,20 +1,44 @@
-import { useQuery } from "@tanstack/react-query"
+import { useMutation, useQuery } from "@tanstack/react-query"
 import { Redirect } from "wouter"
 
 import { getUser } from "../utils/query-user"
 import { FullPageLoader } from "./Loader"
+import { loginToSite } from "../utils/switch-site"
+import { useEffect } from "react"
+import { ThemeKey } from "../types/theme"
+import { getCurrentTheme } from "../utils/current-theme"
 
 interface Props {
   children: React.ReactNode
 }
 
 export const AuthCheck = (props: Props) => {
-  const query = useQuery({ queryKey: ["auth"], queryFn: getUser })
+  const query = useQuery({
+    queryKey: ["auth"],
+    queryFn: getUser,
+  })
 
-  if (query.isLoading) return <FullPageLoader />
+  const loginMutation = useMutation({
+    mutationFn: loginToSite,
+    mutationKey: ["login"],
+  })
+
+  useEffect(() => {
+    if (!query.data?.email) {
```

Comment:
If the user does not have a session yet, we login for them. This is needed for the JWT auth and SSO. We decided to simulate the login for them, so the UX is a bit easier than having to login.

Reply:
Can you point me to where the decision has been made? I'm curious.

Reply:
@WiNloSt here you go: https://metaboat.slack.com/archives/C063Q3F1HPF/p1719509315975669?thread_ts=1719295515.987739&cid=C063Q3F1HPF

Reply:
awesome thanks!